src/components/Breadcrumbs/Breadcrumbs.vue

```diff
@@ -0,0 +1,385 @@
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<docs>
+
+### General description
+
+This component renders a breadcrumb bar. It adjusts to the available width
+by hiding breadcrumbs in a dropdown menu and emits an event when something
+is dropped on a creadcrumb.
+
+### Usage
+
+```vue
+<template>
+	<div>
+		<div class="container container__wide">
+			<Breadcrumbs @dropped="dropped">
+				<Breadcrumb v-for="crumb in breadcrumbs" :title="crumb.title" :path="crumb.path">
+			</Breadcrumbs>
+		</div>
+		<br />
+		<div class="container container__narrow">
+			<Breadcrumbs @dropped="dropped">
+				<Breadcrumb v-for="crumb in breadcrumbs" :title="crumb.title" :path="crumb.path">
+			</Breadcrumbs>
+		</div>
+		<br />
+		<div class="dragme" draggable="true">
+			<span>Drag me onto the breadcrumbs.</span>
+		</div>
+	</div>
+</template>
+
+<script>
+export default {
+	computed: {
+		breadcrumbs() {
+			const path = 'Folder 1/Folder 2/Folder 3/Folder 4'
+			const crumbs = (path === '') ? [] : path.split('/')
+			return [{ title: 'Home', path: '/' }].concat(crumbs.map((crumb, i) => {
+				return {
+					title: crumb,
+					path: '/' + crumbs.slice(0, i + 1).join('/'),
+				}
+			}))
+		},
+	},
+	methods: {
+		dropped(path) {
+			alert('Dropped something on ' + path)
+		},
+	}
+}
+</script>
+<style>
+.container {
+	display: inline-flex;
+}
+
+.container__wide {
+	width: 800px;
+}
+
+.container__narrow {
+	width: 250px;
+}
+
+.dragme {
+	display: block;
+	width: 100px;
+	height: 44px;
+	background-color: var(--color-background-dark);
+}
+</style>
+```
+</docs>
+
+<script>
+import Vue from 'vue'
+import debounce from 'debounce'
+import { Actions } from '../Actions'
+import { ActionRouter } from '../ActionRouter'
+
+export default {
+	name: 'Breadcrumbs',
+	components: {
+		Actions,
+		ActionRouter
+	},
+	props: {
+		rootIcon: {
+			type: String,
+			required: false,
+			default: 'icon-home'
+		}
+	},
+	data: function() {
+		return {
+			hiddenIndices: [],
+			hiddenCrumbs: []
+		}
+	},
+	created() {
+		window.addEventListener('resize', debounce(() => {
+			this.handleWindowResize()
+		}, 100))
+	},
+	mounted() {
+		this.handleWindowResize()
+	},
+	updated() {
+		this.$nextTick(() => {
+			this.handleWindowResize()
+		})
+	},
+	beforeDestroy() {
+		window.removeEventListener('resize', this.handleWindowResize)
+	},
+	methods: {
+		closeActions(e) {
+			// Get the correct element, the events are bound to the <a>
+			if (e.target.closest) {
+				const target = e.target.closest('.crumb .dropdown')
+				// Don't do anything if we leave towards a child element.
```

Comment:
I don't understand what is this part about ?

Reply:
This prevents that the dropdown is immediately closed when you hover from the Actions three-dot icon towards the single Actions. The drag-leave event is bound to the `<a>`, when you leave it, `closeActions` is called, we then need to check if we are still in the Actions menu of have left it completely. I will try to add a better comment for this.